examples/lighting-app/qpg/src/ZclCallbacks.cpp

```diff
@@ -62,6 +64,70 @@ void emberAfPostAttributeChangeCallback(EndpointId endpoint, ClusterId clusterId
             ChipLogError(Zcl, "wrong length for level: %d", size);
         }
     }
+    else if (clusterId == ZCL_COLOR_CONTROL_CLUSTER_ID)
+    {
+        /* ignore several attributes that are currently not processed */
+        if ((attributeId == ZCL_COLOR_CONTROL_REMAINING_TIME_ATTRIBUTE_ID) ||
+            (attributeId == ZCL_COLOR_CONTROL_ENHANCED_COLOR_MODE_ATTRIBUTE_ID) ||
+            (attributeId == ZCL_COLOR_CONTROL_COLOR_MODE_ATTRIBUTE_ID))
+        {
+            return;
+        }
+
+        if ((attributeId != ZCL_COLOR_CONTROL_CURRENT_X_ATTRIBUTE_ID) &&
+            (attributeId != ZCL_COLOR_CONTROL_CURRENT_Y_ATTRIBUTE_ID) &&
+            (attributeId != ZCL_COLOR_CONTROL_CURRENT_HUE_ATTRIBUTE_ID) &&
+            (attributeId != ZCL_COLOR_CONTROL_CURRENT_SATURATION_ATTRIBUTE_ID))
+        {
+            ChipLogProgress(Zcl, "Unknown attribute ID: %" PRIx32, attributeId);
+            return;
+        }
+
+        if (size == sizeof(uint16_t))
+        {
+            XyColor_t xy;
+            if (attributeId == ZCL_COLOR_CONTROL_CURRENT_X_ATTRIBUTE_ID)
+            {
+                xy.x = *static_cast<uint16_t *>(static_cast<void *>(value));
```

Comment:
probably ok for QPG, howevever if this gets copied and pasted it is not endian safe. Consider endian reader/writer or endian macros instead of raw memory casts. That also prevents alignment issues (not sure if QPG cares about those for 16-bit)